src/benchmarks/micro/runtime/JetStream/Poker/poker.cs

```diff
@@ -0,0 +1,366 @@
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace JetStream {
+    public class CardDeck
+    {
+        public static readonly string[] _newDeck = new[] {
+            // Spades
+            "\x1f0a1", "\x1f0a2",  "\x1f0a3",  "\x1f0a4",  "\x1f0a5",
+            "\x1f0a6", "\x1f0a7",  "\x1f0a8",  "\x1f0a9",  "\x1f0aa",
+            "\x1f0ab", "\x1f0ad",  "\x1f0ae",
+            // Hearts
+            "\x1f0b1", "\x1f0b2",  "\x1f0b3",  "\x1f0b4",  "\x1f0b5",
+            "\x1f0b6", "\x1f0b7",  "\x1f0b8",  "\x1f0b9",  "\x1f0ba",
+            "\x1f0bb", "\x1f0bd",  "\x1f0be",
+            // Clubs
+            "\x1f0d1", "\x1f0d2",  "\x1f0d3",  "\x1f0d4",  "\x1f0d5",
+            "\x1f0d6", "\x1f0d7",  "\x1f0d8",  "\x1f0d9",  "\x1f0da",
+            "\x1f0db", "\x1f0dd",  "\x1f0de",
+            // Diamonds
+            "\x1f0c1", "\x1f0c2",  "\x1f0c3",  "\x1f0c4",  "\x1f0c5",
+            "\x1f0c6", "\x1f0c7",  "\x1f0c8",  "\x1f0c9",  "\x1f0ca",
+            "\x1f0cb", "\x1f0cd",  "\x1f0ce"
+        };
+
+        public static readonly string[] _rankNames = new[] {
+            "", "Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "", "Queen", "King"
+        };
+
+        public List<string> _cards;
+
+        public CardDeck ()
+        {
+            this.newDeck();
+        }
+
+        public void newDeck()
+        {
+            // Make a shallow copy of a new deck
+            this._cards = new List<string>(_newDeck);
+        }
+
+        public void shuffle()
+        {
+            this.newDeck();
+
+            for (int index = 52; index != 0;) {
+                // Select a random card
+                var randomIndex = (int)Math.Floor(Compat.random() * index);
```

Comment:
Is the amount of CPU work dependent on the cards that get selected by the `shuffle()` method? If so, we should always choose the same set of cards to allow for [apples to apples comparison](https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#Always-the-same-input-data) over time. Otherwise, the benchmark might report different timings when run multiple times.

```suggestion

            var random = new Random(12345); // const seed to always have the same data generated
            for (int index = 52; index != 0;) {
                // Select a random card
                var randomIndex = (int)Math.Floor(random.NextDouble() * index);
```

Reply:
It wouldn't surprise me if there are subtle performance variances based on the order it comes up with so it makes sense to lock the seed, even though the original benchmark doesn't do it. We were originally using this to try and 1:1 compare with native JS but that's not a reasonable goal when it's in BDN.